aem/events_mgmt_aem/src/main/java/com/adobe/aio/aem/event/management/EventProviderRegistrationService.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package com.adobe.aio.aem.event.management;
+
+import com.adobe.aio.aem.status.StatusSupplier;
+import com.adobe.aio.event.management.model.EventMetadata;
+import com.adobe.aio.event.management.model.Provider;
+
+/**
+ * AEM has to register against Adobe I/O Events API management, in order to be assigned a
+ * providerId, and then to be able to register its event metadata
+ */
+public interface EventProviderRegistrationService extends StatusSupplier {
```

Comment:
@francoisledroff Shouldn't the AEM OSGI bundle transparently map the same interface as our core java SDK?

For example, in this interface we cherry-picked only these two methods `getRegisteredProvider` and `registerEventMetadata`. Any reason for not implementing ALL the methods that we have in [ProviderService](https://github.com/adobe/aio-lib-java/blob/main/events_mgmt/src/main/java/com/adobe/aio/event/management/ProviderService.java) and [RegistrationService](https://github.com/adobe/aio-lib-java/blob/main/events_mgmt/src/main/java/com/adobe/aio/event/management/RegistrationService.java) in two separate OSGI interfaces?

Reply:
All AEM interfaces have a subset of methods. For instance, in [EventPublishService](https://github.com/adobe/aio-lib-java/pull/54/files#diff-3d8a75b0df9fad41d46b1c148420c99e36968d7f8190cf2a073c42b4a235d8f1R18) also, we have one method out of many present in [PublishService](https://github.com/adobe/aio-lib-java/blob/main/events_ingress/src/main/java/com/adobe/aio/event/publish/PublishService.java)